### PR TITLE
beforeAll and afterAll don't emit step test anymore in JUnitRunner

### PR DESCRIPTION
**Pull request by kaatzee**
Fix proposal for #691

### CODE REVIEW

File: junit/jvm/src/test/scala/org/specs2/runner/JUnitRunnerSpec.scala [hunk 1]
```diff
@@ -58,3 +89,17 @@ object LinkedJUnitSpec extends mutable.Specification {
...
+@RunWith(classOf[JUnitRunner])
+class JUnitWithBeforeAfterAllSpecification extends Specification with BeforeAfterAll {
+  def is =
+    s2"""
+
+  one example $ok
+
+"""
+
+  override def beforeAll(): Unit = {}
```
Comment (etorreborre) [inline junit/jvm/src/test/scala/org/specs2/runner/JUnitRunnerSpec.scala]: I would rather write 
```scala
def beforeAll: Unit = ()
```
here to stay consistent with other places in the code.

**Comment by etorreborre** [comment]
This looks like the right thing to do indeed, thanks.

**Comment by etorreborre** [comment]
The only issue is: how do we report a step error? We should probably do a savage `println` in that case.

**Comment by kaatzee** [comment]
According to reporting error in step. I have looked to JUnit code([ParentRunner.java](https://github.com/junit-team/junit4/blob/02c328028b4d32c15bbf0becc9838e54ecbafcbf/src/main/java/org/junit/runners/ParentRunner.java#L390), [EachTestNotifier.java](https://github.com/junit-team/junit4/blob/master/src/main/java/org/junit/internal/runners/model/EachTestNotifier.java#L19)) and also enhanced a little bit my example([RunListenerTest.java](https://github.com/kaatzee/specs2Issues/blob/master/src/test/java/s2i/junitlistener/RunListenerTest.java), [JUnitExecutionListener.java](https://github.com/kaatzee/specs2Issues/blob/master/src/main/java/s2i/junitlistener/JUnitExecutionListener.java)) and it looks like that for exception in beforeAll just calling method `fireTestFailure` with proper description name is enough. In such case description name should be set to `"RunListenerTest.java"` instead of `"step(RunListenerTest.java)"`. Here ([JUnit4TestListener.java](https://github.com/JetBrains/intellij-community/blob/master/plugins/junit_rt/src/com/intellij/junit4/JUnit4TestListener.java#L212) we can see that for example Intellij is prepared for such scenario.
I can try to prepare some fix proposal for this but only after weekend.
Have a nice weekend!

**Comment by kaatzee** [comment]
Replaced by #700.

**Pull request closed without merging**